### PR TITLE
[stable/mongodb-replicaset] Adjusting storage to non alpha/beta if using v1.7 or higher

### PR DESCRIPTION
**Pull request by venezia**
Per https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1 moving storage class setting to `spec.storageClassName`

Also incrementing the version to account for the change

### CODE REVIEW

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 1]
```diff
@@ -166,11 +166,13 @@ spec:
     - metadata:
         name: datadir
         annotations:
+        {{- if lt .Capabilities.KubeVersion.Minor "7" }}
         {{- if .Values.persistentVolume.storageClass | quote }}
           volume.beta.kubernetes.io/storage-class: {{ .Values.persistentVolume.storageClass | quote }}
         {{- else }}
           volume.alpha.kubernetes.io/storage-class: default
         {{- end }}
+        {{- end }}
```
Comment (mgoodness) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml:175]: This whole block can be deleted, as we don't currently support K8s < 1.6.

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 2]
```diff
@@ -179,6 +181,13 @@ spec:
         {{- range .Values.persistentVolume.accessModes }}
           - {{ . | quote }}
         {{- end }}
+        {{- if ge .Capabilities.KubeVersion.Minor "7" }}
+        {{- if .Values.persistentVolume.storageClass }}
+        storageClassName: {{ .Values.persistentVolume.storageClass | quote }}
+        {{- else }}
+        storageClassName: default
+        {{- end }}
+        {{- end }}
```
Comment (mgoodness) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml:190]: Copy the pattern introduced in #1332

**Review by mgoodness — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
This has been superseded by #2013.

Related issue: #1869

**Pull request closed without merging**